File: request2.js
function avg(data){
  var sum = 0;
  var dog_num = 0;
  var list = data["employees"];
  var avg_salary = 0;
  
  for(var i = 0; i < list.length; i++){
    if (list[i]["manager"] == false){
      sum += list[i]["salary"];
      dog_num += 1;
    }
  }
  avg_salary = sum/dog_num;
  console.log(avg_salary); 
}

avg({
"employees":[
{
"name":"John",
"salary":30000,
"manager":false },
{
"name":"Bob",
"salary":60000,
"manager":true },
{
"name":"Jenny",
"salary":50000,
"manager":false },
{
"name":"Tony",
"salary":40000,
"manager":false }
]
});

File: request2.py
def avg(data):
  sum = 0
  dog_num = 0
  #計算字串、列表、字典、元組的長度
  total_people= len(data["employees"])
  for i in range(total_people):
    if (data["employees"][i]["manager"] == False):
      sum += data["employees"][i]["salary"]
      dog_num += 1
  avg_salary = sum/dog_num
  print("avg_salary= %d" %avg_salary)
  


avg({
"employees":[ 
  {
    "name":"John", 
    "salary":30000, 
    "manager":False
  },
  {
    "name":"Bob", 
    "salary":60000, 
    "manager":True
  }, 
  {
    "name":"Jenny", 
    "salary":50000, 
    "manager":False
  }, 
  {
    "name":"Tony", 
    "salary":40000, 
    "manager":False
  } 
]
}) 
